pyomo/contrib/appsi/base.py

```diff
@@ -1463,6 +1460,12 @@ def update(self, timer: HierarchicalTimer = None):
             self.set_objective(pyomo_obj)
         timer.stop('objective')
 
+        # this has to be done after the objective in case the
+        # old objective uses old variables
+        timer.start('vars')
+        self.remove_variables(old_vars)
+        timer.stop('vars')
```

Comment:
Just to make sure I'm understanding, it has to be after constraints for the same reason, right? (It just already was.)

Reply:
I'll update the comment